File: frontend/src/api/userService.js
import api from './axios';

export const userService = {
  getMe: () => api.get('/users/me'),
  updateMe: (data) => api.put('/users/me', data),
  deleteMe: () => api.delete('/users/me'),
  getAllUsers: (page = 0, size = 10) => api.get(`/users?page=${page}&size=${size}`),
  getUserById: (id) => api.get(`/users/${id}`),
};

File: backend/src/main/java/com/freelancing/service/UserService.java
package com.freelancing.service;

import com.freelancing.dto.request.UpdateUserRequest;
import com.freelancing.dto.response.UserResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface UserService {
    UserResponse getCurrentUser(Long userId);
    UserResponse updateCurrentUser(Long userId, UpdateUserRequest request);
    void deactivateAccount(Long userId);
    Page<UserResponse> getAllUsers(Pageable pageable);
    UserResponse getUserById(Long id);
    void banUser(Long id);
    void unbanUser(Long id);
}
